crates/wasmtime/src/func.rs

```diff
@@ -829,30 +680,58 @@ impl Func {
     /// in a "blocking context".
     ///
     /// For more information see the documentation on [asynchronous
-    /// configs](Config::async_support).
+    /// configs](crate::Config::async_support).
     ///
     /// # Panics
     ///
     /// Panics if this is called on a function in a synchronous store. This
     /// only works with functions defined within an asynchronous store.
     #[cfg(feature = "async")]
     #[cfg_attr(nightlydoc, doc(cfg(feature = "async")))]
-    pub async fn call_async(&self, params: &[Val]) -> Result<Box<[Val]>> {
+    pub async fn call_async<T>(
+        &self,
+        mut store: impl AsContextMut<Data = T>,
+        params: &[Val],
+    ) -> Result<Box<[Val]>>
+    where
+        T: Send,
```

Comment:
This is an interesting bound. This is required by internals (notably the `opaque_send()` function). I think it would be possible to make this future conditionally `Send` if desired based on `T` (rather than always requiring `T: Send`), it'll just require some refactoring internally from what the current structure is. I'm not sure that it's all too useful to have that, so I went ahead and just added `T: Send` to the various async methods and if users really want non-`Send` futures with wasmtime we can come back and remove this via other means in the future.